fc/permission_request_test.go

```
@@ -0,0 +1,642 @@
+		WithNameForMultifieldRequest(&PersonName{Given: "Marian", Family: "Reed"}),
+		WithLocationForMultifieldRequest(NewLocation(
+			WithAddressLine1("123/23"),
+			WithPostalCode("23432"))))
+	pr, _ := NewPermissionRequest(
+		WithMultifieldRequestForPermission(multifieldRequest),
+		WithConsentPurposeForPermission(consentPurpose1),
+		WithConsentPurposeForPermission(consentPurpose2),
+		WithCollectionMethodForPermission("cookiePopUp"),
+		WithCollectionLocationForPermission("Can we get a snapshot of where someone is opting in/out here?"),
+		WithPolicyUrlForPermission("http://foo.baz"),
+		WithTermsServiceForPermission("http://foo.tos"))
+	err := validateForPermissionCreate(pr)
+	assert.NoError(t, err)
+}
+
+func TestNewPermissionRequestForCreateWithMultipleConsentPurposeWithoutChannel(t *testing.T) {
+	consentPurpose1 := NewConsentPurpose(
+		WithConsentPurposeId(1),
+		WithConsentPurposeChannel("web"),
+		WithConsentPurposeTtl(365),
+		WithConsentPurposeEnabled(true))
+	consentPurpose2 := NewConsentPurpose(
+		WithConsentPurposeId(2),
+		WithConsentPurposeChannel("phone"),
+		WithConsentPurposeChannel("mobile"),
+		WithConsentPurposeTtl(365),
+		WithConsentPurposeEnabled(true))
+	multifieldRequest, _ := NewMultifieldRequest(
+		WithNameForMultifieldRequest(&PersonName{Given: "Marian", Family: "Reed"}),
+		WithLocationForMultifieldRequest(NewLocation(
+			WithAddressLine1("123/23"),
+			WithPostalCode("23432"))))
+	pr, _ := NewPermissionRequest(
+		WithMultifieldRequestForPermission(multifieldRequest),
+		WithConsentPurposeForPermission(consentPurpose1),
+		WithConsentPurposeForPermission(consentPurpose2),
+		WithCollectionMethodForPermission("cookiePopUp"),
+		WithCollectionLocationForPermission("Can we get a snapshot of where someone is opting in/out here?"),
+		WithPolicyUrlForPermission("http://foo.baz"),
+		WithTermsServiceForPermission("http://foo.tos"))
+	err := validateForPermissionCreate(pr)
+	assert.NoError(t, err)
```

It should throw an error

---

Updated